pepper-apis/dsm-core/src/test/java/org/broadinstitute/dsm/service/participantdata/RgpParticipantDataServiceTest.java

```diff
@@ -0,0 +1,256 @@
+import org.broadinstitute.dsm.util.TestParticipantUtil;
+import org.junit.After;
+import org.junit.AfterClass;
+import org.junit.Assert;
+import org.junit.BeforeClass;
+import org.junit.Test;
+
+
+@Slf4j
+public class RgpParticipantDataServiceTest extends DbAndElasticBaseTest {
+    private static final String instanceName = "rgpservice";
+    private static String esIndex;
+    private static DDPInstanceDto ddpInstanceDto;
+    private static final List<ParticipantDto> participants = new ArrayList<>();
+    private static int participantCounter = 0;
+    private static final List<Integer> fieldSettingsIds = new ArrayList<>();
+
+    @BeforeClass
+    public static void setup() throws Exception {
+        esIndex = ElasticTestUtil.createIndex(instanceName, "elastic/atcpMappings.json",
+                "elastic/atcpSettings.json");
+        ddpInstanceDto = DdpInstanceGroupTestUtil.createTestDdpInstance(instanceName, esIndex);
+    }
+
+    @AfterClass
+    public static void tearDown() {
+        DdpInstanceGroupTestUtil.deleteInstance(ddpInstanceDto);
+        ElasticTestUtil.deleteIndex(esIndex);
+    }
+
+    @After
+    public void deleteParticipantData() {
+        ParticipantDataDao participantDataDao = new ParticipantDataDao();
+        List<ParticipantData> participantDataList =
+                participantDataDao.getParticipantDataByInstanceId(ddpInstanceDto.getDdpInstanceId());
+        participantDataList.forEach(participantData ->
+                participantDataDao.delete(participantData.getParticipantDataId()));
+
+        participants.forEach(participantDto ->
+                TestParticipantUtil.deleteParticipant(participantDto.getParticipantId().orElseThrow()));
+        participants.clear();
+        fieldSettingsIds.forEach(FieldSettingsTestUtil::deleteFieldSettings);
+        fieldSettingsIds.clear();
+    }
+
+    @Test
+    public void getFamilyId() {
+        String participantId = "PTP123";
+        String instanceId = "rgp_family_id";
+        long familyId = 1000;
+        int bookmarkId = 1;
+
+        BookmarkDto bookmarkDto = new BookmarkDto.Builder(familyId, instanceId).withBookmarkId(bookmarkId).build();
+        BookmarkDao bookmarkDao = mock(BookmarkDao.class);
+        when(bookmarkDao.getBookmarkByInstance(instanceId)).thenReturn(Optional.of(bookmarkDto));
+        when(bookmarkDao.updateBookmarkValueByBookmarkId(bookmarkId, familyId)).thenReturn(1);
+        Bookmark bookmark = new Bookmark(bookmarkDao);
+
+        Assert.assertEquals(familyId, RgpParticipantDataService.getNextFamilyId(participantId, bookmark));
+
+        when(bookmarkDao.getBookmarkByInstance(instanceId)).thenThrow(new RuntimeException("Error getting bookmark with instance"));
+        try {
+            RgpParticipantDataService.getNextFamilyId(participantId, bookmark);
+        } catch (RuntimeException e) {
+            Assert.assertTrue(e.getMessage().contains("not found in Bookmark table"));
+        }
+    }
+
+    @Test
```

Comment:
Most of the tests in this file are not new, although they have been tweaked to fit with testing support changes. The following two tests, testBuildParticipantData and testCreateDefaultData, are new, as is much of the associated setup and verification code.